File: java/tibrv-messaging/src/main/java/com/example/messaging/rendezvous/TibrvSettings.java
package com.example.messaging.rendezvous;

/**
 * Holds the three values needed to connect to TIBCO Rendezvous. Together they
 * determine a single communication path (Transport).
 *
 * <ul>
 *   <li>Service : UDP service/port to use (e.g. "7500"). Empty uses the RV default.</li>
 *   <li>Network : network card / multicast address (e.g. ";225.1.1.1"). Usually empty.</li>
 *   <li>Daemon  : rvd daemon address (e.g. "tcp:7500" or "tcp:host:7500").</li>
 * </ul>
 */
public final class TibrvSettings {

    private String service;
    private String network;
    private String daemon;

    /** Initializes every value to an empty string (connect to the default local RV daemon). */
    public TibrvSettings() {
        this("", "", "");
    }

    /**
     * Creates settings from the three explicit values.
     *
     * @param service UDP service/port
     * @param network network value
     * @param daemon  rvd daemon address
     */
    public TibrvSettings(String service, String network, String daemon) {
        this.service = service != null ? service : "";
        this.network = network != null ? network : "";
        this.daemon = daemon != null ? daemon : "";
    }

    /** @return UDP service (port); an empty string uses the RV default. */
    public String getService() {
        return this.service;
    }

    public void setService(String service) {
        this.service = service != null ? service : "";
    }

    /** @return network (NIC / multicast) value; usually an empty string. */
    public String getNetwork() {
        return this.network;
    }

    public void setNetwork(String network) {
        this.network = network != null ? network : "";
    }

    /** @return rvd daemon address (e.g. "tcp:7500"). */
    public String getDaemon() {
        return this.daemon;
    }

    public void setDaemon(String daemon) {
        this.daemon = daemon != null ? daemon : "";
    }
}

File: Messaging/TibrvSettings.cs
namespace com.example.Messaging.Rendezvous
{
    /// <summary>
    /// Holds the three values needed to connect to TIBCO Rendezvous. Together they
    /// determine a single communication path (Transport).
    ///
    /// - Service : UDP service/port to use (e.g. "7500"). Empty uses the RV default.
    /// - Network : network card / multicast address (e.g. ";225.1.1.1"). Usually empty.
    /// - Daemon  : rvd daemon address (e.g. "tcp:7500" or "tcp:host:7500").
    /// </summary>
    public sealed class TibrvSettings
    {
        /// <summary>UDP service (port). An empty string uses the RV default.</summary>
        public string Service { get; set; }

        /// <summary>Network (NIC / multicast) value. Usually an empty string.</summary>
        public string Network { get; set; }

        /// <summary>rvd daemon address (e.g. "tcp:7500").</summary>
        public string Daemon { get; set; }

        /// <summary>
        /// Initializes every value to an empty string (i.e. connect to the default local RV daemon).
        /// </summary>
        public TibrvSettings()
        {
            this.Service = string.Empty;
            this.Network = string.Empty;
            this.Daemon = string.Empty;
        }

        /// <summary>Creates settings from the three explicit values.</summary>
        /// <param name="service">UDP service/port</param>
        /// <param name="network">network value</param>
        /// <param name="daemon">rvd daemon address</param>
        public TibrvSettings(string service, string network, string daemon)
        {
            this.Service = service ?? string.Empty;
            this.Network = network ?? string.Empty;
            this.Daemon = daemon ?? string.Empty;
        }
    }
}
